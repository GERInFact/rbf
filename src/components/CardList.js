import React from "react";
import Card from "./Card";

import "./cardList.css";

const CardList = (props) => {
  const { users } = props;
  return users.length ? (
    <div className="cards">
      {users.map((u, i) => {
        return <Card key={i} id={u.id} name={u.name} email={u.email} />;
      })}
    </div>
  ) : <h2>Loading...</h2>;
};

export default CardList;
