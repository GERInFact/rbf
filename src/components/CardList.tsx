import React from "react";
import Card from "./Card";
import { IUser } from "../containers/App";

import "./cardList.css";

const CardList = ({ users }: { users: Array<IUser> }): JSX.Element => {
  return users.length ? (
    <div className="cards">
      {users.map((u: IUser, i: number) => {
        return <Card key={i} id={u.id} name={u.name} email={u.email} />;
      })}
    </div>
  ) : (
      <h2>Nothing found!</h2>
    );
};

export default CardList;
