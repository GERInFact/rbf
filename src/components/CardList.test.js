import React from "react";
import renderer from "react-test-renderer";
import CardList from "./CardList";

it("expect to render CardList compoenent", () => {
    const mockRobots = [
        {
            id: 1,
            name: "Robo",
            email: "robo@test.com"
        },
        {
            id: 2,
            name: "BoomBaam",
            email: "boombaam@test.com"
        }
    ];

    const component = renderer.create(<CardList users={mockRobots} />);
    expect(component.toJSON()).toMatchSnapshot();
});