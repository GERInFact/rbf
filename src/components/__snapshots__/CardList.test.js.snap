// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList compoenent 1`] = `
<div
  className="cards"
>
  <div
    className="card"
  >
    <div
      className="card_img-container"
    >
      <img
        alt="A random robot face"
        className="img-container_img"
        src="https://robohash.org/1?200x200"
      />
    </div>
    <div
      className="card_info"
    >
      <h3
        className="info_name"
      >
        Robo
      </h3>
      <p
        className="info_email"
      >
        robo@test.com
      </p>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card_img-container"
    >
      <img
        alt="A random robot face"
        className="img-container_img"
        src="https://robohash.org/2?200x200"
      />
    </div>
    <div
      className="card_info"
    >
      <h3
        className="info_name"
      >
        BoomBaam
      </h3>
      <p
        className="info_email"
      >
        boombaam@test.com
      </p>
    </div>
  </div>
</div>
`;
